rpmfile/__init__.py

```diff
@@ -1,44 +1,82 @@
-
 from __future__ import print_function, unicode_literals, absolute_import
-from .headers import get_headers
-import sys
-import io
+
 import gzip
+import io
+import sys
+
+from .headers import get_headers
+
 try:
     import lzma
 except ImportError:
     pass
 import struct
 from rpmfile import cpiofile
-from functools import wraps
 from rpmfile.io_extra import _SubFile
 
 pad = lambda fileobj: (4 - (fileobj.tell() % 4)) % 4
 
+
 class NoLZMAModuleError(NotImplementedError):
     pass
 
+
 class RPMInfo(object):
-    '''
+    """
     Informational class which holds the details about an
     archive member given by an RPM entry block.
-    RPMInfo objects are returned by RPMFile.getmember() and 
+    RPMInfo objects are returned by RPMFile.getmember() and
     RPMFile.getmembers() and are
     usually created internally.
-    '''
+    """
     _new_coder = struct.Struct(b'8s8s8s8s8s8s8s8s8s8s8s8s8s')
 
-    def __init__(self, name, file_start, file_size, initial_offset, isdir):
+    def __init__(self, name, file_start, file_size, initial_offset,
+                 isdir, ino, is_elf):
         self.name = name
-        self.file_start = file_start
-        self.size = file_size
+        self._file_start = file_start
+        self._size = file_size
         self.initial_offset = initial_offset
         self._isdir = isdir
+        self._ino = ino
+        self._is_elf = is_elf
+        self.link_target = None
+
+    @property
+    def file_start(self):
+        if self.link_target is not None:
+            assert self.link_target != self
```

Comment:
What does it mean?

Reply:
it should avoid links to self (circular dependency), added assertion message